### ISSUE TITLE
sshd fails to chroot /var/run/sshd

### ISSUE THREAD

**Issue by marlaw** [corbinlc/GNURootDebian#133]
Installed the testing version (versionCode >= 55 and the versionName >= 0.6.5) and installed openssh-server.

When I try to connect to it the connection gets reset by peer (I tried from both the same device as well as another, on the right ports) the connection gets reset by peer.

I tried to run sshd -d, with debug and the problem is that on connection it tries to chroot into /var/run/sshd and fails to do so. Any idea?

**Comment by marlaw** [comment]
One workaround seems to be to set in /etc/ssh/sshd_config

UsePrivilegeSeparation=no

**Comment by corbinlc** [comment]
Yes, that workaround works for openssh.  You can also switch to using dropbear, which doesn't have this issue (dropbear is already installed for you).  The issue is that open-ssh using chroot and that doesn't work when running in proot.

**Comment by marlaw** [comment]
@corbinlc thank you, shall i close this one then?

**Comment by corbinlc** [comment]
I guess I could create a default sshd_config for people.  That would mostly solve this.  Maybe there is some other way to force that variable.

**Comment by marlaw** [comment]
Either a default sshd_config or as part of the install script you can run something like

`sed -i -- 's/UsePrivilegeSeparation yes/UsePrivilegeSeparation no/g' /etc/ssh/sshd_config`

so proot can't handle chroot I take? or is there a fix for that?

**Comment by Redsandro** [comment]
@corbinlc [commented](https://github.com/corbinlc/GNURootDebian/issues/133#issuecomment-266151487):
> (dropbear is already installed for you)

Really? Where, how?

For me it just installs a new one when `apt-get install dropbear`.

Anyway, I couldn't get it to use public key authorization, so I installed `openssh-server` and this worked using `/root/.ssh/authorized_keys`. Still quite a hazzle to set it up on the phone using the phone.

I'd really :heart: to see an option when installing this app to paste a public key and it will automatically set up ssh to be able to login using that key.

**Comment by AdamMatthewLTM** [comment]
Hi all,

I am currently assisting the developers in sorting relevant issues for GNURoot Debian on GitHub. If the issue discussed is still relevant, could you please reply to this thread and state the issue needs to be resolved and I will attempt to sort it for you ASAP. If the issue is no longer relevant, please reply to this thread and state it has been resolved, and I will suggest this thread be closed.

Furthermore, if a particular issue is considered to be high priority, please state that you are requesting it to be considered as high priority and I will mark it as such.

Thank you all so much for your time and understanding,
-Adam

**Comment by corbinlc** [comment]
We are putting our efforts into a new, but similar, app called UserLAnd.  It is now in beta and it should provide ssh and/or vnc working out of the box.  To join the beta of UserLAnd, please go here:  https://play.google.com/apps/testing/tech.ula
